helper/helper_test.go

```diff
@@ -98,4 +294,234 @@ var _ = Describe("helper tests", func() {
+			}
+
+			helper := newHelperTest(oidBatch, expectedScenario)
+			err := doRestoreAgentInternal(helper)
+			Expect(err).To(BeNil())
+		})
+		It("calls Wait in waitForPlugin doRestoreAgent for single data file", func() {
+			*singleDataFile = true
+			*isResizeRestore = false
+			*tocFile = testTocFile
+
+			// Although pure concept would be to mock TOC file as well, to keep things simpler
+			// let's use real TOC file here
+			writeTestTOC(testTocFile)
+			defer func() {
+				_ = os.Remove(*tocFile)
+			}()
+
+			oidBatch := []oidWithBatch{{100, 0}}
+			expectedScenario := []helperTestStep{
+				{"mock_100_0", true, -1, false, "Some pipe shall be created, out of interest for this test although"},
+			}
+			helper := newHelperTest(oidBatch, expectedScenario)
+
+			err := doRestoreAgentInternal(helper)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Check that plugin command's Wait was acually called and only once
+			Expect(helper.restoreData.waitCount).To(Equal(1))
+		})
+		It("calls waitForPlugin doRestoreAgent for resize and no single data file ", func() {
+			Expect(*singleDataFile).To(Equal(false))
+
+			oidBatch := []oidWithBatch{{100, 0}}
+			expectedScenario := []helperTestStep{
+				{"mock_100_0", true, -1, false, "Some pipe shall be created, out of interest for this test although"},
+			}
+
+			helper := newHelperTest(oidBatch, expectedScenario)
+
+			err := doRestoreAgentInternal(helper)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Check that plugin command's Wait was acually called and only once
+			Expect(helper.restoreData.waitCount).To(Equal(1))
+		})
+		It("calls waitForPlugin doRestoreAgent for reduce cluster and no single data file ", func() {
+			Expect(*singleDataFile).To(Equal(false))
+			*destSize, *origSize = *origSize, *destSize
+
+			oidBatch := []oidWithBatch{{100, 0}}
+			expectedScenario := []helperTestStep{
+				{"mock_100_0", true, -1, false, "Some pipe shall be created, out of interest for this test although"},
+			}
+
+			helper := newHelperTest(oidBatch, expectedScenario)
+
+			err := doRestoreAgentInternal(helper)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Check that plugin command's Wait was acually called and only once
+			Expect(helper.restoreData.waitCount).To(Equal(1))
+		})
+	})
+	Describe("RestoreReader tests", func() {
+		It("waitForPlugin normal completion", func() {
+			test_cmd1 := testPluginCmd{hasProcess_: true}
+			test_reader := new(RestoreReader)
+			test_reader.pluginCmd = &test_cmd1
+
+			err := test_reader.waitForPlugin()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(test_cmd1.waitCount).To(Equal(1))
+
+			// Check that waitForPlugin do nothing when no cmd and/or no process
+			test_cmd2 := testPluginCmd{hasProcess_: false}
+			test_reader.pluginCmd = &test_cmd2
+			err = test_reader.waitForPlugin()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(test_cmd2.waitCount).To(Equal(0))
```

Comment:
may be split it to another test?

Reply:
what it will bring us then?

Reply:
In the "waitForPlugin normal completion" test there are essentially two tests, one with `test_cmd1` and the second with `test_cmd2`, I suggest dividing them into different tests.